docs/developer/architecture/user_customisations.rst

```diff
@@ -0,0 +1,51 @@
+User Customisations and Persistence
+===================================
+
+Context
+-------
+
+MetalK8s comes with a set of tools and services that may need to be configured
+on site. At the same time, we don't want the administrator of the cluster to
+master each and every service of the cluster. We also don't want to allow any
+kind of customisations since it would make the system even more complex to test
+and maintain over the time.
+
+In addition to those services, MetalK8s deployment may have to be adapted
+depending on the architecture of the platform or depending on the different
+use cases running on the top of it.
+
+Be it services or MetalK8s customisations, we need to ensure it is persisted and
+resistant to various type of events such as node reboot, upgrade, downgrade,
+backup, restore
+
+.. _usercusto-requirements:
+
+Requirements
+------------
+
+As a MetalK8s administrator, I have access to a finite list of settings I can
+customise on-site in order to match with my environment specificities:
+
+- list of static users and credentials configured in Dex
+- Integration with an external IDP configuration in Dex
+- Alert notification configuration in Alert Manager
+- Number of replicas for the Prometheus and Alert Manager cluster
+- Other items will appear as we add new customisable features in MetalK8s
+
+Depending on the the setting, impact of the customisation can be at cluster
+level or namespace level (not sure it is needed)
+
+As a MetalK8s administrator, I have a uniq CLI/API way of doing MetalK8s
+customisation, whatever the type of customisation is.
+From a UI, I could have different type of forms in order to make the
+configuration easier depending on the type of setting.
+
+As a MetalK8s administrator, I have access to a documented list of settings I
+can configure in the MetalK8s Operation section.
+
+As a MetalK8s administrator, I can upgrade or downgrade my cluster without
+losing any of the customised settings described above.
+
+As a MetalK8s administrator, when I am doing a backup of my cluster, I backup
+all the customised settings described above and I can restore it when restoring
```

Comment:
> I backup all the customised settings

No, we don't backup anything about user custos, but they are in etcd so even if we lose node they are still available in the etcd cluster. 
For me this section should just be removed

Reply:
My point is to make sure that if we lose the cluster or etcd we have a backup of user custos that we can restore. So for everything in etcd we have the etcd backup but what about BootstrapConfig, Node Objects config, Solution / Environment configs etc ... If everything is already implemented this is great but the requirement remains valid no ?

Reply:
In the backup we have all `/etc/metalk8s` directory and etcd snapshot but currently we only do backup at the end of bootstrap and restore.

The think is that currently we do not have automatic backup. We only backup at the end of the bootstrap and at the end of restoration but that it. That mean if a user do a custos in etcd or bootstrap config, or whatever, after bootstrap (which will always be the case) that will not be in the backup.

Anyway we should often backup.

If you restore with an etcd cluster still alive then you will not lose any tunning in the etcd stuff.
But for all tuning in bootstrap file AND/OR if you lose all etcd cluster we can't guarantee that your backup archive is uptodate with all the tuning you did.

Anyway I'm ok with this requirement just for me the work that need to be done on backuo/restore is not related to this EPIC

Reply:
Requirement is valid and we should keep it. It is a good reason to choose `etcd` as a storage media, since we'll need to back it up if we want to restore after complete disaster (everything lost).
Although we speak a lot about this kind of disaster recovery, I'm curious if this is a real need, and if it is, what's the urgency. Losing a cluster is a lot more problems than just a loss of small configurations, and nothing we can manage on MetalK8s' side.

Reply:
> No, we don't backup anything about user custos, but they are in etcd so even if we lose node they are still available in the etcd cluster.
> For me this section should just be removed

Don't use the design to adjust the requirements.